packages/file_selector/file_selector_platform_interface/lib/src/types/x_type_group/x_type_group.dart

```diff
@@ -1,27 +1,26 @@
 // Copyright 2013 The Flutter Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
+import 'package:meta/meta.dart';
 
 /// A set of allowed XTypes
+@immutable
 class XTypeGroup {
   /// Creates a new group with the given label and file extensions.
   ///
   /// A group with none of the type options provided indicates that any type is
   /// allowed.
-  XTypeGroup({
+  const XTypeGroup({
     this.label,
     List<String>? extensions,
     this.mimeTypes,
     this.macUTIs,
     this.webWildCards,
-  }) : extensions = _removeLeadingDots(extensions);
```

Comment:
I forgot about this. Doing it in the getter as you've done here seems like reasonable price to pay for being able to do const construction, since it's not like we expect a lot of repeated access to the property.

@ditman Does that seem good to you as well?

Reply:
Absolutely